datafusion/sql/src/expr/mod.rs

```diff
@@ -236,14 +236,14 @@ impl<S: ContextProvider> SqlToRel<'_, S> {
                 self.sql_identifier_to_expr(id, schema, planner_context)
             }
 
-            SQLExpr::MapAccess { .. } => {
-                not_impl_err!("Map Access")
-            }
-
             // <expr>["foo"], <expr>[4] or <expr>[4:5]
-            SQLExpr::Subscript { expr, subscript } => {
-                self.sql_subscript_to_expr(*expr, subscript, schema, planner_context)
-            }
+            SQLExpr::CompoundFieldAccess { root, access_chain } => self
```

Comment:
The code to plan CompoundFieldAccess is directly copy/pasted from @goldmedal 's PR
- https://github.com/apache/datafusion/pull/13734